hscontrol/policy/acls_test.go

```diff
@@ -2794,15 +2794,83 @@ func Test_getFilteredByACLPeers(t *testing.T) {
 				},
 			},
 		},
+
+		{
+			name: "subnet-router-with-only-route",
+			args: args{
+				nodes: []*types.Node{
+					{
+						ID:          1,
+						IPAddresses: []netip.Addr{netip.MustParseAddr("100.64.0.1")},
+						Hostname:    "user1",
+						User:        types.User{Name: "user1"},
+					},
+					{
+						ID:          2,
+						IPAddresses: []netip.Addr{netip.MustParseAddr("100.64.0.2")},
+						Hostname:    "router",
+						User:        types.User{Name: "router"},
+						Routes: types.Routes{
+							{
+								NodeID:    2,
+								Prefix:    types.IPPrefix(netip.MustParsePrefix("10.33.0.0/16")),
+								IsPrimary: true,
+								Enabled:   true,
+							},
+						},
+					},
+				},
+				rules: []tailcfg.FilterRule{
+					{
+						SrcIPs: []string{
+							"100.64.0.1/32",
+						},
+						DstPorts: []tailcfg.NetPortRange{
+							{IP: "10.33.0.0/16", Ports: tailcfg.PortRangeAny},
+						},
+					},
+				},
+				node: &types.Node{
+					ID:          1,
+					IPAddresses: []netip.Addr{netip.MustParseAddr("100.64.0.1")},
+					Hostname:    "user1",
+					User:        types.User{Name: "user1"},
+				},
+			},
+			want: []*types.Node{
+				{
+					ID:          2,
+					IPAddresses: []netip.Addr{netip.MustParseAddr("100.64.0.2")},
+					Hostname:    "router",
+					User:        types.User{Name: "router"},
+					Routes: types.Routes{
+						{
+							NodeID:    2,
+							Prefix:    types.IPPrefix(netip.MustParsePrefix("10.33.0.0/16")),
+							IsPrimary: true,
+							Enabled:   true,
+						},
+					},
+				},
+			},
+		},
 	}
+
+	// TODO(kradalby): Remove when we have gotten rid of IPPrefix type
+	prefixComparer := cmp.Comparer(func(x, y types.IPPrefix) bool {
```

Comment:
Could you add this comparer to the util.Comparers?

Reply:
Not possible, that will incur a cyclic dependency due to the IPPrefix type needing to import "types" into "util", but "util" already importing "types". This same issue exists in the route_test.go

Reply:
oh yes 🤦